src/App.js

```diff
@@ -1,101 +1,53 @@
-import { useEffect, useState } from 'react'
-import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
-import { useDispatch, useSelector } from 'react-redux'
-import 'reactjs-popup/dist/index.css'
-import { fetchMovies } from './data/moviesSlice'
-import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
-import Header from './components/Header'
-import Movies from './components/Movies'
-import Starred from './components/Starred'
-import WatchLater from './components/WatchLater'
-import YouTubePlayer from './components/YoutubePlayer'
-import './app.scss'
+import { useEffect } from "react";
+
+import {
+  Routes,
+  Route,
+  useSearchParams,
+  useLocation,
+  useNavigate,
+} from "react-router-dom";
+
+import FavoriteMoviesPage from "./modules/favoriteMovies/FavoriteMovies.page.jsx";
+import PopularMoviesPage from "./modules/popularMovies/PopularMovies.page.jsx";
+import SearchMoviesPage from "./modules/searchMovies/SearchMovies.page.jsx";
+import WatchLaterPage from "./modules/watchLaterMovies/WatchLaterMovies.page.jsx";
+import Header from "./shared/components/header/Header.jsx";
+import "./App.scss";
+import useDebounce from "./shared/hooks/useDebounce.js";
 
 const App = () => {
+  const [searchParams] = useSearchParams();
+  const searchQuery = useDebounce(searchParams.get("q") || "", 800);
+  const location = useLocation();
+  const navigate = useNavigate();
 
-  const state = useSelector((state) => state)
-  const { movies } = state  
-  const dispatch = useDispatch()
-  const [searchParams, setSearchParams] = useSearchParams()
-  const searchQuery = searchParams.get('search')
-  const [videoKey, setVideoKey] = useState()
-  const [isOpen, setOpen] = useState(false)
-  const navigate = useNavigate()
-  
-  const closeModal = () => setOpen(false)
-  
-  const closeCard = () => {
```

Comment:
there were empty function bodies and unused variables in the code. I updated eslint to include these as errors.